utils/dataset_validator.py

```diff
@@ -0,0 +1,116 @@
+from flask_uuid import UUID_RE
+
+DATASET_NAME_LEN_MIN = 1
+DATASET_NAME_LEN_MAX = 100
+
+CLASS_NAME_LEN_MIN = 1
+CLASS_NAME_LEN_MAX = 100
+
+
+def validate(dataset):
+    """Validator for datasets.
+
+    Dataset must have the following structure:
+    {
+        - name (string)
+        - description (string, optional)
+        - classes (list of class dicts)
+            {
+                - name (string)
+                - description (string, optional)
+                - recordings (list of UUIDs)
+            }
+    }
+
+    Complete dataset must contain at least two classes with two recordings in
+    each class.
+
+    Args:
+        dataset: Dataset stored in a dictionary.
+
+    Raises:
+        ValidationException: A general exception for validation errors.
+        IncompleteDatasetException: Raised in cases when one of "completeness"
+            requirements is not satisfied.
+    """
+    if not isinstance(dataset, dict):
+        raise ValidationException("Dataset must be a dictionary.")
+    _check_dict_structure(dataset, ["name", "description", "classes", "public"])
+
+    # Name
+    if "name" not in dataset:
+        raise ValidationException("Dataset must have a name.")
+    if not isinstance(dataset["name"], basestring):
+        raise ValidationException("Class name must be a string.")
+    if not (DATASET_NAME_LEN_MIN < len(dataset["name"]) < DATASET_NAME_LEN_MAX):
+        raise ValidationException("Class name must be between %s and %s characters" %
+                                  (DATASET_NAME_LEN_MIN, DATASET_NAME_LEN_MAX))
+
+    # Description (optional)
+    if "description" in dataset:
+        if not isinstance(dataset["description"], basestring):
```

Comment:
I don't think I like making modifications to a dataset here. This just checks if it has exactly the right format.